src/session/sidebar/row.rs

```diff
@@ -72,223 +73,255 @@ mod imp {
-                Chat::static_type(),
-                Some(&chat_expression),
-                "notification-settings",
-            );
-
-            let session_expression = gtk::ConstantExpression::new(&chat.session());
-
-            let scope_notification_settings_expression = gtk::PropertyExpression::new(
-                Session::static_type(),
-                Some(&session_expression),
-                match chat.type_() {
-                    ChatType::Private(_) | ChatType::Secret(_) => {
-                        "private-chats-notification-settings"
-                    }
-                    ChatType::BasicGroup(_) => "group-chats-notification-settings",
-                    ChatType::Supergroup(data) => {
-                        if data.is_channel {
-                            "channel-chats-notification-settings"
+        let self_ = imp::Row::from_instance(self);
+
+        if let Some(ref item) = item {
+            if let Some(chat) = item.downcast_ref::<Chat>() {
+                self_.timestamp_label.set_visible(true);
+                self_.bottom_box.set_visible(true);
+
+                // Chat properties expressions
+                let title_expression = gtk::PropertyExpression::new(
+                    Chat::static_type(),
+                    gtk::NONE_EXPRESSION,
+                    "title",
+                );
+                let last_message_expression = gtk::PropertyExpression::new(
+                    Chat::static_type(),
+                    gtk::NONE_EXPRESSION,
+                    "last-message",
+                );
+                let unread_count_expression = gtk::PropertyExpression::new(
+                    Chat::static_type(),
+                    gtk::NONE_EXPRESSION,
+                    "unread-count",
+                );
+                let is_pinned_expression = gtk::PropertyExpression::new(
+                    Chat::static_type(),
+                    gtk::NONE_EXPRESSION,
+                    "is-pinned",
+                );
+                let notification_settings_expression = gtk::PropertyExpression::new(
+                    Chat::static_type(),
+                    gtk::NONE_EXPRESSION,
+                    "notification-settings",
+                );
+                let session_expression = gtk::PropertyExpression::new(
+                    Chat::static_type(),
+                    gtk::NONE_EXPRESSION,
+                    "session",
+                );
+
+                // Title label bindings
+                title_expression.bind(&*self_.title_label, "label", Some(chat));
+
+                // Timestamp label bindings
+                let date_expression = gtk::PropertyExpression::new(
+                    Message::static_type(),
+                    Some(&last_message_expression),
+                    "date",
+                );
+                let timestamp_expression = gtk::ClosureExpression::new(
+                    move |args| -> String {
```

Comment:
Not part of the PR, but as we do massive refactorings anyway: This `move` is not needed